modules/apps/site-initializer/site-initializer-extender/site-initializer-extender-test/src/testIntegration/java/com/liferay/site/initializer/extender/internal/test/BundleSiteInitializerTest.java

```diff
@@ -722,19 +745,57 @@ private void _assertCommerceNotificationTemplate(
 			commerceNotificationTemplate.getName());
 	}
 
-	private void _assertCommerceSpecificationProducts() throws Exception {
+	private void _assertCommerceSpecificationProducts1() throws Exception {
 		CPSpecificationOption cpSpecificationOption =
 			_cpSpecificationOptionLocalService.fetchCPSpecificationOption(
 				_serviceContext.getCompanyId(), "test-product-specification-1");
 
 		Assert.assertNotNull(cpSpecificationOption);
+		Assert.assertFalse(cpSpecificationOption.getCPOptionCategoryId() > 0);
 
-		CPDefinition cpDefinition =
+		CPDefinition cpDefinition1 =
 			_cpDefinitionLocalService.
 				fetchCPDefinitionByCProductExternalReferenceCode(
 					"TESTCOMMERCEPRODUCT1", _serviceContext.getCompanyId());
 
-		Assert.assertNotNull(cpDefinition);
+		Assert.assertNotNull(cpDefinition1);
+
+		ProductSpecificationResource.Builder
+			productSpecificationResourceBuilder =
+				_productSpecificationResourceFactory.create();
+
+		ProductSpecificationResource productSpecificationResource =
+			productSpecificationResourceBuilder.user(
+				_serviceContext.fetchUser()
+			).build();
+
+		Page<ProductSpecification> productSpecificationPage =
+			productSpecificationResource.getProductIdProductSpecificationsPage(
+				cpDefinition1.getCProductId(), Pagination.of(1, 10));
+
+		ProductSpecification productSpecification =
+			productSpecificationPage.fetchFirstItem();
+
+		Assert.assertNotNull(productSpecification);
+		Assert.assertEquals(
+			"test-product-specification-1",
+			productSpecification.getSpecificationKey());
```

Comment:
move this logic to a method to assert the relation between products and spectifications using localservice